core/src/main/scala/com/itv/bucky/AmqpClient.scala

```diff
@@ -53,6 +57,17 @@ object AmqpClient extends StrictLogging {
         commandBuilder.toPublishCommand(message)
       }) { publisher }
 
+  def publisherWithHeadersOf[F[_], T](commandBuilder: PublishCommandBuilder[T])(publisher: Publisher[F, PublishCommand])(
```

Comment:
If we updated the existing `Publisher` type and just had one method `publisher`, that by default didn't add any headers to the message, would that mean we'd have to update all client code?